### ISSUE TITLE
[Spree Upgrade] Included vs not included Adjustments in the edit order page (1 broken test)

### ISSUE THREAD

**Issue by luisramos0** [openfoodfoundation/openfoodnetwork#2940]
There are no adjustments showing up on the edit order page in the following spec.
And we need to clarify what are order charges and what are price adjustments and how they map to the data model of order.adjustments.

```
  8) 
    As an administrator
    I want to manage orders
 as an enterprise manager viewing the edit page shows the order charges (non-tax adjustments)
     Failure/Error: expect(page).to have_selector "td", match: :first, text: adjustment.label
       expected to find visible css "td" with text "Shipping" within #<Capybara::Node::Element tag="tbody" path="//HTML[1]/BODY[1]/DIV[1]/DIV[1]/DIV[4]/DIV[1]/DIV[1]/DIV[1]/DIV[2]/DIV[2]/DIV[1]/DIV[1]/FIELDSET[1]/TABLE[1]/TBODY[1]"> but there were no matches
     # ./spec/features/admin/orders_spec.rb:244:in `block (6 levels) in <top (required)>'
     # ./spec/features/admin/orders_spec.rb:242:in `block (5 levels) in <top (required)>'
     # ./spec/features/admin/orders_spec.rb:241:in `block (4 levels) in <top (required)>'
```

and 

```
 9) 
    As an administrator
    I want to manage orders
 as an enterprise manager viewing the edit page shows the order taxes
     Failure/Error:
       within('table.index tbody#price-adjustments') do
         expect(page).to have_selector "td", match: :first, text: "Tax 1"
         expect(page).to have_selector "td.total", text: Spree::Money.new(10)
       end
     
     Capybara::ElementNotFound:
       Unable to find visible css "table.index tbody#price-adjustments"
     # ./spec/features/admin/orders_spec.rb:257:in `block (4 levels) in <top (required)>'
```

The problem will probably be in the order factory used. Some detail must be missing, probably how payment or shipment is setup in the factory order_with_taxes that needs to be adapted to spree 2.

**Comment by luisramos0** [comment]
still 2 errors but not the same specs:
```
7) 
    As an administrator
    I want to manage orders
 as an enterprise manager viewing the edit page shows the order total
     Failure/Error:
       within('fieldset#order-total') do
         expect(page).to have_selector "span.order-total", text: @order.display_total
       end
     
     Capybara::ElementNotFound:
       Unable to find visible css "fieldset#order-total"
     # ./spec/features/admin/orders_spec.rb:251:in `block (4 levels) in <top (required)>'

  8) 
    As an administrator
    I want to manage orders
 as an enterprise manager viewing the edit page shows the order taxes
     Failure/Error:
       within('table.index tbody#price-adjustments') do
         expect(page).to have_selector "td", match: :first, text: "Tax 1"
         expect(page).to have_selector "td.total", text: Spree::Money.new(10)
       end
     
     Capybara::ElementNotFound:
       Unable to find visible css "table.index tbody#price-adjustments"
     # ./spec/features/admin/orders_spec.rb:257:in `block (4 levels) in <top (required)>'
```

**Comment by luisramos0** [comment]
After PR 3286, these are the 3 errors related to OC and distributor:

```
  1) 
    As an administrator
    I want to manage orders
 creating an order with distributor and order cycle
     Failure/Error: expect(page).to have_selector 'h1', text: 'Customer Details'
       expected to find visible css "h1" with text "Customer Details" but there were no matches. Also found "Order # R268713701", which matched the selector but not all filters.
     # ./spec/features/admin/orders_spec.rb:81:in `block (2 levels) in <top (required)>'

  2) 
    As an administrator
    I want to manage orders
 displays error when incorrect distribution for products is chosen
     Failure/Error: targetted_select2(value, options)
     
     Capybara::ElementNotFound:
       Unable to find visible css "#s2id_order_distributor_id"
     # ./spec/support/request/web_helper.rb:163:in `select2_select'
     # ./spec/features/admin/orders_spec.rb:116:in `block (2 levels) in <top (required)>'

 5) 
    As an administrator
    I want to manage orders
 as an enterprise manager creating an order with distributor and order cycle
     Failure/Error: expect(page).to have_selector 'td', text: product.name
       expected to find visible css "td" with text "Product #29 - 516" but there were no matches. Also found "Subtotal :", "$0.00", "", "TOTAL ORDER :", "$0.00", "", which matched the selector but not all filters.
     # ./spec/features/admin/orders_spec.rb:327:in `block (3 levels) in <top (required)>'
```

**Comment by luisramos0** [comment]
The spec related to "non-tax adjustments" is now passing, but it shouldnt be. I think the spec is green because the spec is not picking up the adjustments, so no validation is being done. But the spec should see the adjustment and validate the adjustment is present on the page.
I think this is now a false positive.

So, there is one broken spec in this issue and one false positive (green spec) that I believe also needs to be fixed.

**Comment by luisramos0** [comment]
I see this issue is being addressed in #3194

**Comment by luisramos0** [comment]
This will not be fixed as part of #3194 
It needs some investigation, the broken spec (order-charges) is not showing the tax of the product because the taxrate is marked as included_in_price and the current adjustments table . is not showing adjustments included in the line items, only order.adjustments. we may decide to change the test or change the view to show adjustments included in the price (order.tax_adjusments instead of order.adjusments for example would show the taxrate)
